test/suites/dev-tanssi/registrar/test_registrar_proxy.ts

```diff
@@ -0,0 +1,123 @@
+            polkadotJs = context.polkadotJs();
+        });
+
+        it({
+            id: "E01",
+            title: "Can add registrar proxy and use it",
+            test: async function () {
+                // Setup proxy
+                const delegate = charlie.address;
+                const registrar_proxy = 6;
+                const delay = 0;
+                const tx = polkadotJs.tx.proxy.addProxy(delegate, registrar_proxy, delay);
+                await context.createBlock([await tx.signAsync(alice)]);
+
+                const events = await polkadotJs.query.system.events();
+                const ev1 = events.filter((a) => {
+                    return a.event.method == "ProxyAdded";
+                });
+                expect(ev1.length).to.be.equal(1);
+
+                const proxies = await polkadotJs.query.proxy.proxies(alice.address);
+                expect(proxies.toJSON()[0]).to.deep.equal([
+                    {
+                        delegate: charlie.address,
+                        proxyType: "Registrar",
+                        delay: 0,
+                    },
+                ]);
+
+                // Use proxy
+                await context.createBlock();
+                const currentSesssion = await polkadotJs.query.session.currentIndex();
+                const sessionDelay = await polkadotJs.consts.registrar.sessionDelay;
+                const expectedScheduledOnboarding =
+                    BigInt(currentSesssion.toString()) + BigInt(sessionDelay.toString());
+
+                const emptyGenesisData = () => {
+                    const g = polkadotJs.createType("TpContainerChainGenesisDataContainerChainGenesisData", {
+                        storage: [
+                            {
+                                key: "0x636f6465",
+                                value: "0x010203040506",
+                            },
+                        ],
+                        name: "0x436f6e7461696e657220436861696e2032303030",
+                        id: "0x636f6e7461696e65722d636861696e2d32303030",
+                        forkId: null,
+                        extensions: "0x",
+                        properties: {
+                            tokenMetadata: {
+                                tokenSymbol: "0x61626364",
+                                ss58Format: 42,
+                                tokenDecimals: 12,
+                            },
+                            isEthereum: false,
+                        },
+                    });
+                    return g;
+                };
+                const containerChainGenesisData = emptyGenesisData();
+
+                const tx2 = polkadotJs.tx.proxy.proxy(
+                    alice.address,
+                    null,
+                    polkadotJs.tx.registrar.register(2002, containerChainGenesisData)
+                );
+                const tx3 = polkadotJs.tx.registrar.markValidForCollating(2002);
```

Comment:
Isnt it sufficient for the test that we test only that the genesis data gets set? the rest is made with sudo